### PR TITLE
Add gradle compatibility section to readme

### PR DESCRIPTION
**Pull request by StefMa**
Just added a nicer table for Gradle compatibility to the readme...

bildschirmfoto 2018-02-03 um 19 08 16

### CODE REVIEW

**Comment by passsy** [comment]
👍 

Should be added for all versions`0.4.0+`

**Comment by StefMa** [comment]
We decided recently that we want support gradle 4.0 and up. I think 0.8.0 supports also 3.2 and below on Java libraries. But I think we have to make a cut and focus on current gradle versions.

Tested it back to 0.4.0 which gradle versions are supporterd are just maintenance overhead which can be avoided.

File: README.md [hunk 1]
```diff
@@ -65,6 +63,10 @@ $ ./gradlew clean build bintrayUpload -PbintrayUser=BINTRAY_USERNAME -PbintrayKe
 
 More info on the available properties and other usages in the [Github Wiki](https://github.com/novoda/bintray-release/wiki).
 
+## Gradle compatibility
+| bintray-release version | Required Gradle version for | Notes                                        |
```
Comment (blundell) [inline README.md]: > Required Gradle version for

I don't really understand this? Can you explain what it is saying sorry.

Ah maybe I get it. Would a better title be:

| Gradle (_wrapper_) version |
| ------ |
| for Java: 4.0 |
| for Android: 4.1 |

Comment (tasomaniac) [reply README.md]: What about this?

| bintray-release version | Java projects | Android projects | 
| --- | --- | --- | 
| 0.8.0 | Gradle 4.0+ | Gradle 4.1+ | 

**Notes: ** Currently Gradle 4.5 doesn't work with Android libraries

Comment (blundell) [reply README.md]: yes 💯

Comment (blundell) [reply README.md]: should we also write what version of the android gradle plugin is needed?

Comment (StefMa) [reply README.md]: I would not do that.
If we start using that "pattern" we have to
* always make sure to take a look which android gradle plugin requires which gradle version (which isn't our responsibility)
* more or less copy the information from the android documentation in our readme

This is just maintenance overhead.

Another example:
Imagine a user have a project with the andorid gradle plugin 2.3.3 and are interested in this plugin.
The see that our plugin requires 4.1 gradle. So he updated it's gradle version to 4.1. and everything works (should works 😄 )
But if we say "android gradle plugin 3.x" he assume that it only works if we update it's grade plugin. Which isn't needed for the bintray-release plugin.

**Comment by tasomaniac** [comment]
@StefMa what about mentioning older `bintray-release` versions? That would be helpful if a user is stuck with Gradle 3.x versions.

**Comment by StefMa** [comment]
Hmm.. I wouldn't do that.
I would update the table only from now on.

Create a changelog for older versions doesn't make sense either :D

**Pull request merged**